### PR TITLE
Siege and assault fixes

### PR DESCRIPTION
**Pull request by TylerS1066**
This PR is on behalf of eirik.

It includes just the edits made to siege and assault, but not the ones for 1.13 support (or siege boss-bar).

### CODE REVIEW

**Comment by TylerS1066** [comment]
@eirikh1996

File: modules/Movecraft/src/main/java/net/countercraft/movecraft/warfare/assault/AssaultTask.java [hunk 1]
```diff
@@ -76,6 +76,7 @@ public void run() {
                 for (x = beaconX + 1; x < beaconX + 4; x++)
                     for (z = beaconZ + 1; z < beaconZ + 4; z++)
                         w.getBlockAt(x, y, z).setType(Material.BEDROCK);
+<<<<<<< HEAD
```
[the remark this answers is not in the document]
Comment (TylerS1066) [reply modules/Movecraft/src/main/java/net/countercraft/movecraft/warfare/assault/AssaultTask.java]: I believe I fixed this in https://github.com/APDevTeam/Movecraft/pull/96/commits/2887a03be4603a50c3a1dca49fda60c9af5c22fa

**Comment by TylerS1066** [comment]
@cccm5 Requested changes made, please review again.

File: modules/Movecraft/src/main/java/net/countercraft/movecraft/listener/BlockListener.java [hunk 2]
```diff
@@ -387,9 +387,12 @@ private void processAssault(EntityExplodeEvent e){
 
                 // whether or not you actually destroyed the block, add to damages
                 long damages = assault.getDamages() + Settings.AssaultDamagesPerBlock;
-                //if (damages < assault.getMaxDamages()) {
+                if (damages < assault.getMaxDamages()) {
```
Comment (cccm5) [inline modules/Movecraft/src/main/java/net/countercraft/movecraft/listener/BlockListener.java]: There is still an edge case that can cause the damages to exceed the maximum. Instead, just pass the minimum of `damages` and `assault::getMaxDamages` to `assault::setDamages`. This allows the removal of this if else clause and avoids any possible overflow.

**Pull request merged**